python/lsst/display/firefly/footprints.py

```diff
@@ -0,0 +1,202 @@
+        spans = footprint.getSpans()
+        scoords = [(s.getY()-y0, s.getX0()-x0, s.getX1()-x0) for s in spans]
+        fpbbox = footprint.getBBox()
+        corners = [(c.getX()-x0,c.getY()-y0) for c in fpbbox.getCorners()]
+        fpxll.append(corners[0][0])
+        fpyll.append(corners[0][1])
+        fpxur.append(corners[2][0])
+        fpyur.append(corners[2][1])
+        peaks = footprint.getPeaks()
+        pcoords = [(p.getFx()-x0, p.getFy()-y0) for p in peaks]
+        parentid = record.getParent()
+        nchild = record.get('deblend_nChild')
+        if (parentid == 0):
+            familylist.append(recordid)
+            if (nchild > 0):
+                # blended parent
+                categorylist.append('blended parent')
+            else:
+                # isolated
+                categorylist.append('isolated')
+        else:
+            # deblended child
+            familylist.append(parentid)
+            categorylist.append('deblended child')
+        spanlist.append(scoords)
+        peaklist.append(pcoords)
+
+    for i in range(len(inputvotable.array)):
+        row = inputvotable.array[i]
+        startlist = [row.item(0)[k] for k in range(len(row))]
+        startlist.insert(insert_column, familylist[i])
+        startlist.insert(insert_column+1, categorylist[i])
+        startlist = startlist + [spanlist[i], peaklist[i],
+                            fpxll[i], fpyll[i], fpxur[i], fpyur[i]]
+        outtable.array[i] = tuple(startlist)
+
+    outtable.infos.append(Info(name='contains_lsst_footprints', value='true'))
+    outtable.infos.append(Info(name='contains_lsst_measurements', value='true'))
+    outtable.infos.append(Info(name='FootPrintColumnNames',
+                               value='id;footprint_corner1_x;footprint_corner1_y;' +
+                              'footprint_corner2_x;footprint_corner2_y;spans;peaks'))
+    outtable.infos.append(Info(name='pixelsys', value='zero-based'))
+    # Check whether the coordinates are included and are valid
+    if (('coord_ra' in sourceTable.colnames)
+        and ('coord_dec' in sourceTable.colnames)
+        and np.isfinite(sourceTable['coord_ra']).any()
+        and np.isfinite(sourceTable['coord_dec']).any()):
+        coord_column_string = 'coord_ra;coord_dec;EQ_J2000'
+    elif (('base_SdssCentroid_x' in sourceTable.colnames)
+        and ('base_SdssCentroid_y' in sourceTable.colnames)
+        and np.isfinite(sourceTable['base_SdssCentroid_x']).any()
+        and np.isfinite(sourceTable['base_SdssCentroid_y']).any()):
+        coord_column_string = 'base_SdssCentroid_x;base_SdssCentroid_y;ZERO_BASED'
+    elif (('base_NaiveCentroid_x' in sourceTable.colnames)
+        and ('base_NaiveCentroid_y' in sourceTable.colnames)
+        and np.isfinite(sourceTable['base_NaiveCentroid_x']).any()
+        and np.isfinite(sourceTable['base_NaiveCentroid_y']).any()):
+        coord_column_string = 'base_NaiveCentroid_x;base_NaiveCentroid_y;ZERO-BASED'
```

Comment:
Longer-term we should try to get the slots transfered out of afw.table so you can just use `slot_Centroid_x` and `slot_Centroid_y`.